website/static/js/licensePicker.js

```diff
@@ -0,0 +1,226 @@
+var OTHER_LICENSE = siteLicenses.OTHER_LICENSE;
+
+var template = require('raw!templates/license-picker.html');
+
+/**
+ * @class LicensePicker: Knockout.js view model for project license selection
+ *
+ * @param {string} saveUrl: url to make save requests to
+ * @param {string} saveMethod: HTTP method for save request (one of 'GET', 'POST', 'PUT', 'PATCH')
+ * @param {string} saveLicenseKey: key to use for save request payload
+ * @param {license} license: optional instantiation license
+ * @property {ko.observable<boolean>} previewing:
+ * @property {ko.observable<license>} savedLicense:
+ * @property {ko.observable<license>} selectedLicense:
+ * @property {license[]} licenses:
+ * @property {ko.observable<string>} notification:
+ * @property {ko.observable<boolean>} error:
+ *
+ * @type license
+ * @property {string} name: human readable name
+ * @property {string} text: full text of license
+ * @property {string} id: unique identifier
+ **/
+var LicensePicker = oop.extend(ChangeMessageMixin, {
+    constructor: function(saveUrl, saveMethod, saveLicenseKey, license, readonly) {
+        this.super.constructor.call(this);
+
+        var self = this;
+
+        var user = $osf.currentUser();
+
+        self.saveUrl = saveUrl || '';
+        self.saveMethod = saveMethod || 'POST';
+        self.saveLicenseKey = saveLicenseKey || 'license';
+
+        self.previewing = ko.observable(false);
+
+        self.licenses = licenses;
+
+        license = license || DEFAULT_LICENSE;
+
+        self.savedLicense = ko.observable(license);
+        self.savedLicenseName = ko.pureComputed(function() {
+            return self.savedLicense().name;
+        });
+        self.savedLicenseId = ko.computed(function() {
+            return self.savedLicense().id;
+        });
+
+        self.selectedLicense = ko.observable(DEFAULT_LICENSE);
+        self.selectedLicenseUrl = ko.pureComputed(function() {
+            return self.selectedLicense().url;
+        });
+        /** ;
+         * Needed to track selected/saved license in the list of licenses
+         **/
+        self.selectedLicenseId = ko.computed({
+            read: function() {
+                return self.selectedLicense().id;
+            },
+            write: function(id) {
+                self.selectedLicense(self.licenses.filter(function(l) {
+                    return l.id === id;
+                })[0]);
+            }
+        });
+        self.selectedLicenseId(license.id);
+
+        self.Year = ko.observable(license.Year || new Date().getFullYear()).extend({
+            required: true,
+            pattern: {
+                params: /^\d{4}\s*$/,
+                message: 'Please specify a valid year.'
+            },
+            max: {
+                params: new Date().getFullYear(),
+                message: 'Future years are not allowed'
+            }
+        });
+        self['Copyright Holders'] = ko.observable(license['Copyright Holders'] || '').extend({
```

Comment:
Can we get rid of these magic strings? These label names should be represented as a mapping of storage-friendly identifiers (e.g. `copyright_holders`) to human-readable labels (e.g. 'Copyright Holders').

That way, if we need to change the human-readable labels, we only need to do so in one place.